### PR TITLE
[PBDEV-1639] fix predicate to detect OSD pod fault events collectly

### PR DESCRIPTION
**Pull request by cheolho-kang**
**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/latest/Contributing/development-flow/#commit-structure).
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/latest/Contributing/development-flow/#submitting-a-pull-request)
- [ ] [Pending release notes](https://github.com/rook/rook/blob/master/PendingReleaseNotes.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.

### CODE REVIEW

File: pkg/operator/ceph/nvmeof_recoverer/nvmeofstorage/controller.go [hunk 1]
```diff
@@ -126,6 +126,9 @@ func add(mgr manager.Manager, r reconcile.Reconciler) error {
 					return false
 				}
 				if isOSDPod(newPod.Labels) && isPodDead(oldPod, newPod) {
+					if newPod.DeletionTimestamp != nil {
```
Comment (myoungwon) [inline pkg/operator/ceph/nvmeof_recoverer/nvmeofstorage/controller.go:130]: 언제 발생하나요?

Comment (cheolho-kang) [reply pkg/operator/ceph/nvmeof_recoverer/nvmeofstorage/controller.go:130]: controller 코드 중
```
func (r *ReconcileNvmeOfStorage) cleanupOSD(namespace string, deviceInfo cephv1.FabricDevice) {
	// Delete the OSD deployment that is in CrashLoopBackOff
	err := k8sutil.DeleteDeployment(
		r.opManagerContext,
		r.context.Clientset,
		namespace,
		osd.AppName+"-"+deviceInfo.OsdID,
	)
```
이 함수에 의해 pod를 삭제하게되면, 이과정도 pod의 상태변화로 간주하고 reconciler가 또 호출될 수 있습니다. 따라서 삭제중인 pod의 상태변화는 필터링하기 위해 이와같이 추가하였습니다

Comment (myoungwon) [reply pkg/operator/ceph/nvmeof_recoverer/nvmeofstorage/controller.go:130]: comment 추가해 주세요

Comment (cheolho-kang) [reply pkg/operator/ceph/nvmeof_recoverer/nvmeofstorage/controller.go:130]: 넵 추가하였습니다

**Pull request merged**